File: project-euler/006/problem006.swift
#!/usr/bin/env xcrun swift
extension Int {
  var squared: Int {
    return self * self
  }
}

let sumOfTheSquares = (0...100).map { $0 * $0 }.reduce(0, +)
let squareOfTheSum = (0...100).reduce(0, +).squared

print(squareOfTheSum - sumOfTheSquares)

File: project-euler/006/problem006.rs
fn square(n: i64) -> i64 {
    n * n
}

fn main() {
    let sum_of_the_squares: i64 = (0..101)
                                  .map(|x| x * x)
                                  .sum();

    let square_of_the_sum = square((0..101).sum());


    println!("{}", square_of_the_sum - sum_of_the_squares);
}